### PR TITLE
Completed Link Class

### PR DESCRIPTION
**Pull request by Samir2003**
Finished link and its subclasses as well as revisions to player.

### CODE REVIEW

File: src/model/data.cc [hunk 1]
```diff
@@ -5,3 +5,30 @@ Data::Data(int strength, char label, Position position, bool boosted,
...
+Position Data::getPosition() const{
+  return position;
+}
+
+int Data::getStrength() const{
+  return strength;
+}
+
+void Data::reveal(){
+  revealed = !revealed;
```
Comment (JacobIm28) [inline src/model/data.cc:29]: Should probably just set to true to be safe in case it's called multiple times

File: src/model/data.cc [hunk 2]
```diff
@@ -5,3 +5,30 @@ Data::Data(int strength, char label, Position position, bool boosted,
...
+}
+
+void Data::boost(){
```
Comment (JacobIm28) [inline src/model/data.cc:32]: Same as for reveal

File: src/model/data.cc (hunk 1 above)
Comment (Samir2003) [reply src/model/data.cc:29]: I thought you may need to hide a link? But I guess not.

File: src/model/virus.cc [hunk 3]
```diff
@@ -5,3 +5,30 @@ Virus::Virus(int strength, char label, Position position, bool boosted,
...
+  return position;
+}
+
+int Virus::getStrength() const{
+  return strength;
+}
+
+void Virus::reveal(){
+  revealed = !revealed;
+}
```
Comment (gnishiwaki) [inline src/model/virus.cc:30]: Im pretty sure a link cannot be unrevealed

**Pull request merged**